### PR TITLE
Upstream a good default keymap for the Ferris

### PR DESCRIPTION
**Pull request by pierrechevalier83**
Upstream a default keymap for the Ferris that uses different behaviours when keys are tapped or held to make for a fully functional keymap on only 34 keys.

It is meant to be a good starting point from which a functional and comfortable keymap for one's specific can be generated incrementally.

There will be a learning curve to learning this keymap as comfort was prioritized over backward compatibility with existing keyboard layouts.

## Description

The readme.md file explains the reasoning behind this keymap.
A short python script (json2crab.py) is used to format the output of the qmk configurator into something that can be read comfortably in a text editor.
The keymap itself is uploaded as a reasonably formatted json file to allow both workflows of editing the keymap directly in a text editor or editing it through the qmk configurator.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/ferris/keymaps/json2crab.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
```
Comment (drashna) [inline keyboards/handwired/ferris/keymaps/json2crab.py:1]: Is this necessary?  

I believe that we do have `qmk json2c` that will handle this, as well.

Comment (pierrechevalier83) [reply keyboards/handwired/ferris/keymaps/json2crab.py:1]: I may have missed something, but I thought json2c only converted from json to c code. What this script is doing is going from unformatted json to formatted json that is more human readable (formatting by half rows of 5 keys and lining up the columns).

Compared to checking out unformatted json from the configurator,
This allows:
* better git diffs (smaller lines)
* easier to read the json without loading it in the configurator
* possible to hand-edit the keymap for small changes without using the configurator

See `default/keymap.json` for an example of output from that script (input is a single line of json coming from the configurator)

If I'm missing a formatting option of json2c, please let me know :)

Comment (pierrechevalier83) [reply keyboards/handwired/ferris/keymaps/json2crab.py:1]: Hi @drashna , any comment on the questions above?

**Comment by pierrechevalier83** [comment]
I'm not seeing any more actionable comment. Is this PR ready to be merged by any chance?

File: keyboards/handwired/ferris/keymaps/default/readme.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+### Layer 0: Base layer
+
+![Layer 0](https://i.imgur.com/HjNHUPL.png)
+
+On tapping the keys, our base layer is qwerty with space on the right homing thumb and backspace on the left homing thumb.
+
+In this layer, the non-homing-thumb positions have 0 and 1. I recommend modifying this to some frequently accessed shortcut such as copy/paste, previous/next tab or anything that makes most sense in your own workflow. O and 1 are place-holders and make it easy to troubleshoot that all keys are working properly before soldering in the switches.
+The reason I recommend convenience shortcuts instead of more commonly used keys like tab or meta is that unhoming of the thumbs was a frequent source of typos for me when I used more than one thumb key frequently in the context of typing.
+
+Despite being missing on this layer, "meta", "tab", "esc" and such are accessible from any othe layer: see Layer 7.
```
Comment (noroadsleft) [inline keyboards/handwired/ferris/keymaps/default/readme.md:26]:
```suggestion
Despite being missing on this layer, "meta", "tab", "esc" and such are accessible from any other layer: see Layer 7.
```

**Comment by pierrechevalier83** [comment]
> Everything else looks good to me.

Thank you! I committed your typo fix :)

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**